src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -0,0 +1,218 @@
+        public DHTSensor(int pin, DhtType dhtType)
+        {
+            this.pin = pin;
+            this.dhtType = dhtType;
+        }
+
+        /// <summary>
+        /// Start a reading
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if read is successfull, otherwise <c>false</c>.
+        /// </returns>
+        public bool ReadData()
+        {
+            using (GpioController controller = new GpioController())
+            {                
+                // Set the max value for waiting micro second
+                // 27 = debug
+                // 99 = release
+                byte waitMS = 99;
+                #if DEBUG
+                    waitMS = 27;
+                 #endif
+                controller.OpenPin(pin);
+
+                DelayMS delayMS= new DelayMS();
+                PinValue lststate = PinValue.High; 
+                uint counter = 0; 
+                byte j = 0, i; 
+                for (i = 0; i < 5; i++) 
+                    dht11_val[i] = 0; 
+
+                // write on the pin
+                controller.SetPinMode(pin, PinMode.Output);
+                controller.Write(pin, PinValue.Low); 
+                //wait 18 milliseconds
+                delayMS.Wait(18); 
+                controller.Write(pin, PinValue.High); 
+                delayMS.Wait(0.02); 
+                controller.SetPinMode(pin, PinMode.Input);
+                for (i = 0; i < MAX_TIME; i++) 
+                { 
+                    counter = 0; 
+                    while (controller.Read(pin) == lststate){ 
+                        counter++; 
+                        // This wait about 1 micro second
+                        // No other way to do it for such a precision
+                        for(byte wt=0; wt< waitMS; wt++)
+                            ; 
+                        if (counter == MAX_WAIT) 
+                            break; 
+                    } 
+                    lststate = controller.Read(pin); 
+                    if (counter == MAX_WAIT)                     
+                        break; 
+                    
+                    // top 3 transistions are ignored   
+                    if ((i >= 4) && (i % 2 == 0)){ 
+                        dht11_val[j / 8] <<= 1; 
+                        if (counter>16) 
+                            dht11_val[j / 8] |= 1; 
+                        j++;                         
+                    } 
+                } 
+
+                if ((j >= 40) && (dht11_val[4] == ((dht11_val[0] + dht11_val[1] + dht11_val[2] + dht11_val[3]) & 0xFF))) 
+                { 
+                    if ((dht11_val[0] == 0) && (dht11_val[2] == 0)) 
+                        LastRead = false;                        
+                    else
+                        LastRead = true;
+
+                } else
+                    LastRead = false;
+                
+                controller.ClosePin(pin);
```

Comment:
I think it might make sense to open the pin in the constructor and close in the dispose since I wouldn't expect users to change how the devices are connected on the regular basis (same for controller)